Sources/SwiftSyntax/Syntax.swift

```diff
@@ -320,22 +269,219 @@ extension Syntax {
+  public var lastToken: TokenSyntax? {
+    return base.lastToken
+  }
+
+  /// The absolute position of the starting point of this node. If the first token
+  /// is with leading trivia, the position points to the start of the leading
+  /// trivia.
+  public var position: AbsolutePosition {
+    return base.position
+  }
+
+  /// The absolute position of the starting point of this node, skipping any
+  /// leading trivia attached to the first token syntax.
+  public var positionAfterSkippingLeadingTrivia: AbsolutePosition {
+    return base.positionAfterSkippingLeadingTrivia
+  }
+
+  /// The absolute position where this node (excluding its trailing trivia)
+  /// ends.
+  public var endPosition: AbsolutePosition {
+    return base.endPosition
+  }
+
+  /// The absolute position where this node's trailing trivia ends
+  public var endPositionAfterTrailingTrivia: AbsolutePosition {
+    return base.endPositionAfterTrailingTrivia
+  }
+
+  /// The textual byte length of this node including leading and trailing trivia.
+  public var byteSize: Int {
+    return base.byteSize
+  }
+
+  /// The length this node takes up spelled out in the source, excluding its
+  /// leading or trailing trivia.
+  public var contentLength: SourceLength {
+    return base.contentLength
+  }
+
+  /// The leading trivia of this syntax node. Leading trivia is attached to
+  /// the first token syntax contained by this node. Without such token, this
+  /// property will return nil.
+  public var leadingTrivia: Trivia? {
+    return base.leadingTrivia
+  }
+
+  /// The trailing trivia of this syntax node. Trailing trivia is attached to
+  /// the last token syntax contained by this node. Without such token, this
+  /// property will return nil.
+  public var trailingTrivia: Trivia? {
+    return base.trailingTrivia
+  }
+
+  /// The length this node's leading trivia takes up spelled out in source.
+  public var leadingTriviaLength: SourceLength {
+    return base.leadingTriviaLength
+  }
+
+  /// The length this node's trailing trivia takes up spelled out in source.
+  public var trailingTriviaLength: SourceLength {
+    return base.trailingTriviaLength
+  }
+
+  /// The length of this node including all of its trivia.
+  public var totalLength: SourceLength {
+    return base.totalLength
+  }
+
+  /// When isImplicit is true, the syntax node doesn't include any
+  /// underlying tokens, e.g. an empty CodeBlockItemList.
+  public var isImplicit: Bool {
+    return base.isImplicit
+  }
+
+  /// The textual byte length of this node exluding leading and trailing trivia.
+  public var byteSizeAfterTrimmingTrivia: Int {
+    return base.byteSizeAfterTrimmingTrivia
+  }
+
+  /// The root of the tree in which this node resides.
+  public var root: Syntax {
+    return base.root
+  }
+
+  /// Passes to a closure every present token node that is part of this node.
+  public func forEachToken(_ receiver: (TokenSyntax)->()) {
```

Comment:
I should have raised this concern earlier, but I'm wary of adding `forEach`-style methods without an equivalent `Sequence`-type. Could we, instead of exposing `forEachToken`, add a `TokenSequence` type that performs the same duty? That gives callers a lot more freedom, and more importantly, allows them to compose with operations like `map()`.

```
for token in node.tokens {
  // we don't need to capture in the closure here, which could save allocations
}

let text = node.tokens.map { $0.text }.joined()
```

Reply:
Good point, much better! Done.